src/main/java/org/mule/extension/api/introspection/metadata/MetadataEnrichableModel.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.api.introspection.metadata;
+
+import org.mule.extension.api.annotation.param.metadata.Content;
+import org.mule.extension.api.annotation.param.metadata.MetadataKeyParam;
+import org.mule.extension.api.introspection.ParameterModel;
+import org.mule.extension.api.metadata.MetadataResolver;
+
+import java.util.Optional;
+
+public interface MetadataEnrichableModel
+{
+
+    /**
+     * @return the {@link MetadataResolverFactory} required to instantiate the
+     * {@link MetadataResolver} associated to this model.
+     */
+    MetadataResolverFactory getMetadataResolverFactory();
+
+    /**
+     * @return the parameter annotated with {@link MetadataKeyParam}, if one is present
+     */
+    Optional<ParameterModel> getMetadataKeyParameter();
+
+    /**
+     * @return the parameter annotated with @Content, if one is present
+     */
+    Optional<ParameterModel> getContentParameter();
+
+
+    /**
+     * Returns true if the operation has a {@link Content} parameter and a
```

Comment:
change "true" and "false" to {@code true} and {@code false}. Remember that this javadoc will change a lot after separating the MetadataResolver into separate concerns